File: web/web-frontend/src/app/model/test/operation/update-test.model.ts
import {Path} from "../../infrastructure/path/path.model";
import {JsonUtil} from "../../../utils/json.util";
import {TestModel} from "../test.model";

export class UpdateTestModel implements Serializable<UpdateTestModel> {

    oldPath:Path;
    testModel:TestModel;

    constructor(oldPath: Path, testModel: TestModel) {
        this.oldPath = oldPath;
        this.testModel = testModel;
    }

    deserialize(input: Object): UpdateTestModel {
        this.oldPath = Path.deserialize(input["oldPath"]);
        this.testModel = input['testModel'];

        return this;
    }

    serialize() {
        return "" +
            '{' +
            '"oldPath":' + JsonUtil.serializeSerializable(this.oldPath) + ',' +
            '"testModel":' + JsonUtil.serializeSerializable(this.testModel) +
            '}'
    }
}

File: model/src/main/java/com/testerum/model/manual/operation/UpdateTestModel.kt
package com.testerum.model.manual.operation

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.model.test.TestModel

data class UpdateTestModel @JsonCreator constructor(
        @JsonProperty("oldPath") val oldPath: Path,
        @JsonProperty("testModel") val testModel: TestModel)